src/constructor_helpers.hpp

```diff
@@ -4,18 +4,40 @@
 #include "phes_base.h"
 #include "kml.h"
 
+// Finds intersections of a polygon with a given latitude
 vector<double> find_polygon_intersections(double lat, vector<GeographicCoordinate> &polygon);
+
+// Checks if a point is within any of the given polygons
 bool check_within(GeographicCoordinate point, vector<vector<GeographicCoordinate>> polygons);
+
+// Reads country data from a file and populates the country names
 vector<vector<vector<GeographicCoordinate>>> read_countries(string filename, vector<string>& country_names);
+
+// Gets adjacent cells between two points
 ArrayCoordinate* get_adjacent_cells(ArrayCoordinate point1, ArrayCoordinate point2);
+
+// Checks if the edge between two points meets a certain threshold
 bool is_edge(ArrayCoordinate point1, ArrayCoordinate point2, Model<char>* model, ArrayCoordinate offset, int threshold);
+
+// Checks if the edge between two points is a dam wall
 bool is_dam_wall(ArrayCoordinate point1, ArrayCoordinate point2, Model<short>* DEM, ArrayCoordinate offset, double wall_elevation);
```

Comment:
I think most of these comments are just rewriting the function signature - I agree some more documentation would be useful but I don't think this is adding much